projects/uma/index.js

```diff
@@ -1,60 +1,259 @@
-const { request, gql } = require("graphql-request");
-const sdk = require('@defillama/sdk');
+const sdk = require("@defillama/sdk");
 const { sumTokens } = require("../helper/unwrapLPs");
 const { requery } = require("../helper/requery");
+const { transformPolygonAddress } = require("../helper/portedTokens");
+const { getBlock } = require("../helper/getBlock");
 
-const endpoint = 'https://api.thegraph.com/subgraphs/name/umaprotocol/mainnet-contracts'
-const query = gql`
-query get_tvl($block: Int) {
-  financialContracts(
-    block: { number: $block }
-  ) {
-    address
-    collateralToken{
-      address
-    }
+const ethLspCreators = [
+  "0x0b8de441B26E36f461b2748919ed71f50593A67b",
+  "0x60F3f5DDE708D097B7F092EFaB2E085AC0a82F42",
+  "0x31C893843685f1255A26502eaB5379A3518Aa5a9",
+  "0x9504b4ab8cd743b06074757d3B1bE3a3aF9cea10",
+];
+const polygonLspCreators = [
+  "0x3e665D15425fAee14eEF53B9caaa0762b243911a",
+  "0x62410e96a2ceB4d66824346e3264d1D9107a0aBE",
+  "0x5Fd7FFF20Ee851cD7bEE72fB3C6d324e4C104c9f",
+];
+
+// Captures TVL for EMP contracts on Ethereum
+async function ethEmp(timestamp, block) {
+  const abi = {
+    inputs: [],
+    name: "collateralCurrency",
+    outputs: [
+      {
+        internalType: "contract IERC20",
+        name: "",
+        type: "address",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  };
+
+  const balances = {};
+  const logs = await sdk.api.util.getLogs({
+    target: "0x3e532e6222afe9Bcf02DCB87216802c75D5113aE",
+    topic: "NewContractRegistered(address,address,address[])",
+    keys: ["topics"],
+    fromBlock: 9937650,
+    toBlock: block,
+  });
+  const collaterals = await sdk.api.abi.multiCall({
+    calls: logs.output.map((poolLog) => ({
+      target: `0x${poolLog[1].slice(26)}`,
+    })),
+    block,
+    abi,
+  });
+  await requery(collaterals, "ethereum", block, abi);
+  await sumTokens(
+    balances,
+    collaterals.output
+      .filter((t) => t.output !== null)
```

Comment:
can we remove this? in some cases our rpc will fail and then we should crash the entire script to avoid writing wrong data to storage

Reply:
The commit uses the contract creator events instead of the registry so it is essentially filtering out the contracts we can't calculate the collateral for and gets us to the same number. Let me know if that works.